view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/ViewGeneratorLauncher.java

```diff
@@ -41,13 +42,24 @@ public StreamsBuilder buildTopology(
 
     Config jobConfig = getJobConfig(properties);
 
-    String inputTopic = jobConfig.getString(INPUT_TOPIC_CONFIG_KEY);
+    List<String> inputTopics = jobConfig.getStringList(INPUT_TOPIC_CONFIG_KEY);
     String outputTopic = jobConfig.getString(OUTPUT_TOPIC_CONFIG_KEY);
 
-    KStream<?, ?> inputStream = inputStreams.get(inputTopic);
-    if (inputStream == null) {
-      inputStream = streamsBuilder.stream(inputTopic, Consumed.with(Serdes.String(), null));
-      inputStreams.put(inputTopic, inputStream);
+    KStream<String, Object> mergeStream = null;
+
+    for (String topic : inputTopics) {
+      KStream<String, Object> inputStream = (KStream<String, Object>) inputStreams.get(topic);
+
+      if (inputStream == null) {
+        inputStream = streamsBuilder.stream(topic, Consumed.with(Serdes.String(), null));
```

Comment:
@laxmanchekka Shall we go ahead here as well as Avro serde instead of String? So, far it was hard-coded key desrilizer as String

Reply:
yes. we need to change this as enricher is producing avro based key

Reply:
Done

Reply:
apologies for misleading on this one here. trace-enricher is producing string based key only. so, we can't move to schema registry based serde